tests/lib/cmsis_dsp/basicmath/src/f32.c

```diff
@@ -5,18 +5,18 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <zephyr/dsp.h>
 #include <zephyr/ztest.h>
 #include <zephyr/kernel.h>
 #include <stdlib.h>
-#include <arm_math.h>
 #include "../../common/test_common.h"
 
 #include "f32.pat"
 
 #define SNR_ERROR_THRESH	((float32_t)120)
 #define REL_ERROR_THRESH	(5.0e-5)
 
-static void test_arm_add_f32(
+static void test_zdsp_add_f32(
```

Comment:
This is very confusing -- a CMSIS-DSP test testing ZDSP?

With the introduction of the "Zephyr Math subsystem" (aka. ZDSP), there should be a dedicated testsuite for it (e.g. `tests/subsys/{math,dsp}/basicmath`).

I acknowledge that having both `tests/subsys/{math,dsp}/basicmath` and `tests/lib/cmsis_dsp/basicmath` is far from ideal since they are basically testing the same thing.

My suggestion is to move the `basicmath` (and other "migrated" API in the future) test to `tests/subsys/{math,dsp}/basicmath`.

p.s. should settle on either `math` or `dsp`.